shortlister/controller.py

```diff
@@ -66,12 +68,32 @@ def edit_criteria_select(self, k=None):
         
     def edit_score_confirm(self,k=None):
         """Confirm changes to score"""
-        
         self.view.view_update(self.current_criterion.name,self.current_criterion.scores[int(k)])
         update_applicant_score(self.current_applicant,self.current_criterion,int(k))
         self.view.view_applicant_details(self.current_applicant)
         self.options = self.options_applicant_detail
 
+    def switch_prev_applicant(self,k=None):
+        """shows details of the previous applicant in the shortlist"""
+        i = self.shortlist.applicants.index(self.current_applicant)
```

Comment:
Instead of saving `self.current_applicant` (i.e. instance of applicant), you should store the _index_ of the current applicant. Then you do not have to constantly search for the applicant in the list (that's what `.index()` is doing). You know exactly which applicant you're looking at. Then get the applicant object from the shortlist applicants list as and when you need it e.g. when passing to a view.